app/_features/event/components/image-cropper.tsx

```diff
@@ -0,0 +1,128 @@
+import { Cropper, ReactCropperElement } from 'react-cropper';
+import {
+  useDisclosure,
+  Modal,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Button,
+} from '@nextui-org/react';
+import { Dispatch, createRef, useCallback, useEffect } from 'react';
+
+export interface ImageState {
+  imageBlob: Blob | null;
+  imagePreview: string | null;
+}
+
+export type ActionType =
+  | { type: 'ConfirmCrop'; payload: { blob: Blob; preview: string } }
+  | { type: 'PickFile'; payload: Blob }
+  | { type: 'Reset' };
+interface ImageCropperModalProps {
+  updateImageData: Dispatch<ActionType>;
+  imageData: ImageState;
+}
+
+export function ImageCropperModal({
+  imageData,
+  updateImageData,
+}: ImageCropperModalProps) {
+  const { isOpen, onOpen, onClose, onOpenChange } = useDisclosure();
+
+  const cropperRef = createRef<ReactCropperElement>();
+
+  const getCropData = useCallback(async () => {
+    if (typeof cropperRef.current?.cropper !== 'undefined') {
+      if (!cropperRef.current) console.log('cropperRef.current is undefined');
+
+      const getCanvasBlob = (
+        canvas: HTMLCanvasElement
+      ): Promise<Blob | null> => {
+        return new Promise(function (resolve) {
+          canvas.toBlob(function (blob) {
+            resolve(blob);
+          }, 'image/jpeg');
+        });
+      };
+
+      const newBlob = await getCanvasBlob(
+        cropperRef.current.cropper.getCroppedCanvas()
+      );
+
+      return newBlob;
+    }
+  }, [cropperRef]);
+
+  const openModal = useCallback(() => {
+    onOpen();
+  }, [onOpen]);
+
+  useEffect(() => {
+    document.addEventListener('open:image-cropper', openModal);
+  });
+
+  const onCancel = useCallback(() => {
+    updateImageData({ type: 'Reset' });
+    onClose();
+  }, [onClose, updateImageData]);
+
+  const onConfirm = useCallback(() => {
+    const newBlob = getCropData();
+
+    newBlob.then((blob) => {
+      if (blob) {
+        updateImageData({
+          type: 'ConfirmCrop',
+          payload: { blob: blob, preview: URL.createObjectURL(blob) },
+        });
+      } else console.log('blob is null');
```

Comment:
Remove this else console.log()

Reply:
I've removed the logging https://github.com/inlivedev/inlive-room/pull/130/commits/06de4972dc0b8b36258652c019bd4a9acd9245ff#diff-d0ca42e2c8f5450a47f15205e8d83111bd45cee06f0be0ff3915367d6d4c6988L78-L80